### ISSUE TITLE
New SEGV during a --profile

### ISSUE THREAD

**Issue by MasterDuke17** [MoarVM/MoarVM#1071]
Profiling `for ^50 { my $cmd = run « cat input.txt », :out; $cmd.out.close; }`, where `input.txt` is a 130m file, causes a segv. Using fewer iterations it sometimes completes successfully.
```
dan@hermes:~/p6$ perl6-gdb-m --profile -e 'for ^50 { my $cmd = run « cat input.txt », :out; $cmd.out.close; }'
================================================================================================
This is Rakudo Perl 6 running in the GNU debugger, which often allows the user to generate useful back-
traces to debug or report issues in Rakudo, the MoarVM backend or the currently running code.

This Rakudo version is 2019.03.1.47.g.76.b.6416.d.0 built on MoarVM version 2019.03.35.gd.15906711,
running on ubuntu (18.10.Cosmic.Cuttlefish) / linux (17.Ubuntu.SMP.Fri.Feb.8.00.06.57.UTC.2019)

Type `bt full` to generate a backtrace if applicable, type `q` to quit or `help` for help.
------------------------------------------------------------------------------------------------
Reading symbols from /home/dan/Source/perl6/install/bin/moar...done.
Starting program: /home/dan/Source/perl6/install/bin/moar --execname=/home/dan/Source/perl6/install/bin/perl6-gdb-m --libpath=/home/dan/Source/perl6/install/share/nqp/lib --libpath=/home/dan/Source/perl6/install/share/perl6/lib --libpath=/home/dan/Source/perl6/install/share/perl6/runtime /home/dan/Source/perl6/install/share/perl6/runtime/perl6.moarvm --profile -e for\ \^50\ \{\ my\ \$cmd\ =\ run\ «\ cat\ input.txt\ »,\ :out\;\ \$cmd.out.close\;\ \}
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7ffff6f8c700 (LWP 6302)]
[New Thread 0x7ffff4a80700 (LWP 6303)]
[New Thread 0x7fffeffff700 (LWP 6304)]
[New Thread 0x7fffef7fe700 (LWP 6305)]
[Detaching after fork from child process 6306]
[New Thread 0x7fffeeffd700 (LWP 6307)]
[Detaching after fork from child process 6308]
[Detaching after fork from child process 6309]
[Detaching after fork from child process 6310]
...
[Detaching after fork from child process 6356]
Writing profiler output to profile-1553041272.4357007.html

Thread 1 "moar" received signal SIGSEGV, Segmentation fault.
0x00007ffff79a8297 in MVM_coerce_smart_stringify (tc=0x55555555acb0, obj=0x55555834ed50, res_reg=0x555557e4bdd8) at src/core/coerce.c:289
289         ss = REPR(obj)->get_storage_spec(tc, STABLE(obj));
(gdb) bt
#0  0x00007ffff79a8297 in MVM_coerce_smart_stringify (tc=0x55555555acb0, obj=0x55555834ed50, res_reg=0x555557e4bdd8) at src/core/coerce.c:289
#1  0x00007ffff7993bcf in MVM_interp_run (tc=tc@entry=0x55555555acb0, initial_invoke=0x5555585cae20, invoke_data=0x555557e4bdd8) at src/core/interp.c:906
#2  0x00007ffff7a868f5 in MVM_vm_run_file (instance=0x55555555a260, filename=<optimized out>) at src/moar.c:444
#3  0x0000555555555791 in main (argc=9, argv=0x7fffffffdd78) at src/main.c:302
(gdb
```

**Comment by dogbert17** [comment]
Problem has possibly been fixed with commit 423a9cc7d922615a1c43f3ac53c4ec9904bf6192

**Comment by MasterDuke17** [comment]
I tested a bunch of runs of 50 iterations and one run of 200 iterations and profiles were created without any segfaults. Looks good, @timo++, @dogbert17++
